### PR TITLE
End Utf8JsonReader token on EndObject and starting Depth

### PR DESCRIPTION
**Pull request by ManiacalSheep**
- Catch and log JsonException in LavaNode instead of JsonReaderException
- Default buffer size to 2048
- LavaNode event JSON document parsing structure fixes
- TrackEndReason being sent as string instead of char, refactor to be static const instead of enum

### CODE REVIEW

File: src/Converters/LavaTrackConverter.cs [hunk 1]
```diff
@@ -23,64 +23,87 @@ internal sealed class LavaTrackConverter : JsonConverter<LavaTrack> {
 
         try
         {
+            var isNearingEndOfPayload = false;
```
Comment (Yucked) [inline src/Converters/LavaTrackConverter.cs]: this seems redundant

File: src/Enums/TrackEndReason.cs
Comment (Yucked) [inline src/Enums/TrackEndReason.cs:1]: Why is this deleted?

File: src/LavaNode.cs [hunk 2]
```diff
@@ -479,7 +476,7 @@ await _webSocketClient.DisposeAsync()
 
                             await OnTrackStart.Invoke(new TrackStartEventArg {
                                 GuildId = guildId,
-                                EncodedTrack = document.GetProperty("encodedTrack").GetString()
+                                EncodedTrack = document.GetProperty("track").GetProperty("encoded").GetString()
```
Comment (Yucked) [inline src/LavaNode.cs]: This should just be a var after event case

File: src/LavaTrack.cs [hunk 3]
```diff
@@ -34,7 +34,7 @@ public class LavaTrack {
     /// 
     /// </summary>
     [JsonPropertyName("uri"), JsonInclude]
-    public string Url { get; internal init; }
+    public string Uri { get; internal init; }
```
Comment (Yucked) [inline src/LavaTrack.cs]: Old version of Victoria still has it has Url, let's keep it that way

File: src/WebSocket/EventArgs/TrackEndEventArg.cs [hunk 4]
```diff
@@ -19,5 +17,5 @@ public struct TrackEndEventArg {
     /// <summary>
     /// 
     /// </summary>
-    public TrackEndReason Reason { get; internal init; }
+    public string Reason { get; internal init; }
```
Comment (Yucked) [inline src/WebSocket/EventArgs/TrackEndEventArg.cs]: Use the enums

File: src/Constants/TrackEndReason.cs
Comment (Yucked) [inline src/Constants/TrackEndReason.cs:1]: Remove this file. Keep the old enums

File: src/Converters/LavaTrackConverter.cs [hunk 5]
```diff
@@ -21,51 +21,72 @@ internal sealed class LavaTrackConverter : JsonConverter<LavaTrack> {
...
+                }
+                else if (IsProp(ref reader, "position"))
+                {
+                    position = reader.GetInt64();
+                }
             }
         }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.ToString());
```
Comment (Yucked) [inline src/Converters/LavaTrackConverter.cs]: Bad practice. Remove Console.WriteLine and try catch

File: src/LavaNode.cs [hunk 6]
```diff
@@ -555,10 +552,13 @@ await _webSocketClient.DisposeAsync()
                 }
             }
         }
-        catch (Exception exception) {
-            _logger.LogError(exception is JsonReaderException
-                ? "Please increase buffer size in configuration."
-                : $"{exception.Message} {exception}");
+        catch (JsonException jsonException)
```
Comment (Yucked) [inline src/LavaNode.cs]: Use pattern match to handle different error message.

```cs
catch (Exception exception) {
 _logger.LogError(exception is JsonException ? "Increase buffer size" : "normal message");
}
```

but there should be no need for this, going to replace buffer with memory stream

**Review by Yucked — CHANGES_REQUESTED** [review]

File: src/Converters/LavaTrackConverter.cs (hunk 1 above)
Comment (ManiacalSheep) [reply src/Converters/LavaTrackConverter.cs]: i'm not catching why, sorry.  what's redunant about it?

File: src/LavaNode.cs (hunk 2 above)
Comment (ManiacalSheep) [reply src/LavaNode.cs]: not following on this one.  what should be a var where?

Comment (Yucked) [reply src/LavaNode.cs]: Track.Encoded should be stored in a var similar to guildId

File: src/Converters/LavaTrackConverter.cs (hunk 1 above)
Comment (Yucked) [reply src/Converters/LavaTrackConverter.cs]: https://github.com/Yucked/Victoria/blob/v6/src/Converters/LavaTracksPropertyConverter.cs

Should follow something similar to this

File: src/LavaNode.cs (hunk 2 above)
Comment (ManiacalSheep) [reply src/LavaNode.cs]: is it not? it's being assigned to EncodedTrack

Comment (ManiacalSheep) [reply src/LavaNode.cs]: nevermind, i think i get it

File: src/Converters/LavaTrackConverter.cs (hunk 1 above)
Comment (ManiacalSheep) [reply src/Converters/LavaTrackConverter.cs]: this one isn't clicking for me, sorry

**Review by Yucked — APPROVED** [review]
